### PR TITLE
[FIX]&[DEV-812]&[DEV-754]&[DEV-751]&[DEV-815]&[DEV-800] link mail broken html tags & only send mail for userData with mail

### PR DESCRIPTION
**Pull request by Krysh90**
#### Pre-Release

- [x] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [x] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/user/models/kyc/kyc-process.service.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class KycProcessService {
       const initiateData = await this.spiderService.initiateIdentification(userData.id, identType);
       userData.spiderData = await this.updateSpiderData(userData, initiateData);
     }
-    if (status === KycStatus.MANUAL)
+    if (status === KycStatus.MANUAL && userData.mail)
```
Comment (davidleomay) [inline src/user/models/kyc/kyc-process.service.ts:55]: @Yannick1712 add a log if there is no mail.

File: src/payment/models/buy/buy.controller.ts [hunk 2]
```diff
@@ -148,7 +149,11 @@ export class BuyController {
     if (dto.amount > frickAmountLimit || dto.currency.name === 'USD') {
       // amount > 9k => Frick || USD => Frick
       account = this.getMatchingAccount(Config.bank.frick.accounts, dto.currency.name, fallBackCurrency);
-    } else if (dto.currency.name === 'EUR' && buy.bankAccount.sctInst) {
+    } else if (
+      dto.currency.name === 'EUR' &&
+      buy.bankAccount.sctInst &&
+      (buy.user.userData.kycStatus === KycStatus.COMPLETED || buy.user.userData.kycStatus === KycStatus.MANUAL)
```
Comment (davidleomay) [inline src/payment/models/buy/buy.controller.ts:155]: Use `KycCompleted` method from user-data.entity.ts

File: src/user/models/kyc/kyc-process.service.ts [hunk 3]
```diff
@@ -52,12 +52,15 @@ export class KycProcessService {
...
-    if (status === KycStatus.MANUAL)
+    if (status === KycStatus.MANUAL && userData.mail) {
       await this.mailService.sendTranslatedMail({
         userData: userData,
         translationKey: 'mail.kyc.success',
         params: {},
       });
+    } else if (status === KycStatus.MANUAL && !userData.mail) {
+      console.error(`Failed to send KYC completion mail for user data ${userData.id}: user has no email`);
+    }
```
Comment (davidleomay) [inline src/user/models/kyc/kyc-process.service.ts:63]: The following seems more readable to me 🤔 

```ts
if (status === KycStatus.MANUAL) {
  if (userData.mail) {
    await this.mailService.sendTranslatedMail({
      userData: userData,
      translationKey: 'mail.kyc.success',
      params: {},
    });
  } else  {
    console.error(`Failed to send KYC completion mail for user data ${userData.id}: user has no email`);
  }
}
```

**Review by davidleomay — APPROVED** [review]

**Review by DFXAdmin — APPROVED** [review]

**Review by DFXAdmin — APPROVED** [review]
das brauch ich unbedingt

**Review by davidleomay — APPROVED** [review]

**Pull request merged**